runtime/compiler/codegen/J9AheadOfTimeCompile.cpp

```diff
@@ -478,13 +479,13 @@ J9::AheadOfTimeCompile::initializeCommonAOTRelocationHeader(TR::IteratedExternal
             cpIndexOrData = static_cast<uintptrj_t>(callSymRef->getCPIndex());
             }
 
-         pgRecord->setInlinedSiteIndex(reloTarget, inlinedSiteIndex);
-         pgRecord->setConstantPool(reloTarget, reinterpret_cast<uintptrj_t>(owningMethod->constantPool()));
-         pgRecord->setCpIndex(reloTarget, cpIndexOrData);
-         pgRecord->setRomClassOffsetInSharedCache(reloTarget, romClassOffsetInSharedCache);
-         pgRecord->setClassChainIdentifyingLoaderOffsetInSharedCache(reloTarget, classChainIdentifyingLoaderOffsetInSharedCache);
-         pgRecord->setClassChainForInlinedMethod(reloTarget, classChainOffsetInSharedCache);
-         pgRecord->setMethodIndex(reloTarget, methodIndex);
+         pRecord->setInlinedSiteIndex(reloTarget, inlinedSiteIndex);
```

Comment:
Did you mean to go through this sequence of `pmgRecord` -> `pgRecord` -> `pRecord` ? I know they're all the same shape at the moment, but I think if you want to share this code, you should probably define a common base `TR_RelocationRecordX` type (just as before this commit you were using `TR_RelocationRecordProfiledGuard` for both class and method guards) and use that here, rather than picking one of the specific types. Or perhaps better to replicate this code across the three relo types as you did for `ValidateStaticField` and `ValidateClass` which also have the same shape. I don't normally like duplicated code, but if you have trouble thinking up an appropriate common name it may be the right choice. It would be nice if we could be consistent in approach across all the relocation record types, too :) .

Reply:
> Did you mean to go through this sequence of pmgRecord -> pgRecord -> pRecord ? 

I did, in order to facilitate the use of `git blame`. I wanted to keep the commits logically distinct.

Reply:
ok